### PR TITLE
Painter partition

### PR DESCRIPTION
**Pull request by NEEHITGOYAL**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

Added solution to this problem using binary search approach to reduce time complexity and also used dynamic array
### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#1068

### CODE REVIEW

**Comment by NEEHITGOYAL** [comment]
kindly review and merge this pr @Kajol-Kumari @SKAUL05 @plazzy99 @HarshCasper

File: C-Plus-Plus/cp/painter_partition.cpp [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+
+int main()
+{
+	int n,k,t;
+	long long int sum=0;
+	cin>>n>>k>>t;
+	int* arr=new int[n];
+	for(int i=0;i<n;i++)
+    {
```
Comment (vatsalkesarwani12) [inline C-Plus-Plus/cp/painter_partition.cpp]: refractor the code with proper indentation.

File: C-Plus-Plus/cp/painter_partition.cpp [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+50
+Explanation
+The first painter can paint the first board in 5 units of time and the second painter 
+will take 50 units of time to paint the second board. Since both can paint simultaneously,
+ the total time required to paint both the boards is 50.
+*/
+
+#include<iostream>
+#include <bits/stdc++.h>
+
```
Comment (vatsalkesarwani12) [inline C-Plus-Plus/cp/painter_partition.cpp]: remove all the unwanted line space

**Review by vatsalkesarwani12 — CHANGES_REQUESTED** [review]

**Comment by NEEHITGOYAL** [comment]
@plazzy99 done identation and removed unwanted spaces

**Review by vatsalkesarwani12 — APPROVED** [review]
👍🏻

**Pull request merged**